strawberryfields/engine.py

```diff
@@ -213,7 +213,7 @@ def _run_program(self, prog, **kwargs):
             list[Command]: commands that were applied to the backend
         """
 
-    def _run(self, program, *, compile_options=None, **kwargs):
+    def _run(self, program, *, args, compile_options, **kwargs):
```

Comment:
See comment in `program.py` about using `params` instead of `args`

Reply:
So `args` always needs to be passed here? i.e., `_run(prog, args={}, compile_options={})`? Previously, `compile_options` had a default value `None`. Curious what the reason for this change is.

Reply:
I think default values are not very useful for "internal use only" methods, especially here since `BaseEngine._run` is only called by `LocalEngine.run`, which provides its own default values anyway.

Depending on how the remote engines work it might also be a good idea to merge the BaseEngine and LocalEngine classes.